scripts/evaluate_chunk_size.py

```diff
@@ -0,0 +1,148 @@
+import os
+import uuid
+import time
+
+import chromadb
+from openai import OpenAI
+
+
+EMBEDDING_COLLECTION_NAME = "learn"
+chunk_sizes = ["paragraph", "half_a_paragraph", "quarter_a_paragraph", "sentence"]
+
+chromadb_client = chromadb.HttpClient(
+    host=os.getenv("CHROMADB_HOST", "chromadb"),
+    port=8000,
+    ssl=False,
+    headers=None,
+    settings=chromadb.config.Settings(),
+    tenant=chromadb.config.DEFAULT_TENANT,
+    database=chromadb.config.DEFAULT_DATABASE,
+)
+
+openai_client = OpenAI(api_key=os.getenv("OPENAI_API_KEY"))
+chromadb_client = chromadb.HttpClient(
+    host=os.getenv("CHROMADB_HOST", "chromadb"),
+    port=8000,
+    ssl=False,
+    headers=None,
+    settings=chromadb.config.Settings(),
+    tenant=chromadb.config.DEFAULT_TENANT,
+    database=chromadb.config.DEFAULT_DATABASE,
+)
+
+# Questions to ask about the text
+f_query = open("../data/RAG_chunk_evaluation_questions.txt", "r")
+my_queries = f_query.read().split("\n")
+f_query.close()
+
+unique_file_id = str(int(time.time()))
+
+for chunk_size_desc in chunk_sizes:
+
+    print(chunk_size_desc)
+
+    # a file to store responses
+    save_path = "evaluate_chunk_size_results"
+    if not os.path.exists(save_path):
+        os.makedirs(save_path)
+    f_response = open(
+        save_path + "/response_" + unique_file_id + "_" + chunk_size_desc + ".txt", "w"
```

Comment:
`nit` most of Python engineers would find f-string more readable and maintainable

f"{save_path}/response_{unique_file_id}_..."

Reply:
`nit` `os.makedirs()` is used to create directories recursively, but in our case, we create only one directory. Therefore, `os.makedir()` is more appropriate, and it can prevent collaborators from thinking the operation will create multiple directories recursively.

Reply:
That is so much easier to read!